### PR TITLE
Added tests for duplicates in changelog and removed duplicates

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR copies the test for duplicate entries from ESMValTool and removes duplicates that have been found by it.

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [ ] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
Why is this failing now?

```
Need type annotation for "ALLOWED_EXCEPTIONS" (hint: "ALLOWED_EXCEPTIONS: List[<type>] = ...")
```

There are 100s of other tests withouth type annotations...

File: tests/unit/test_docs.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""Tests for the documentation."""
```
Comment (valeriupredoi) [inline tests/unit/test_docs.py:1]: cheers Manu! Here's a general point about this test module: I am happy to see a start of actually testing the documentation outside the CI test that just builds it, but I'd be happier to see a dedicated test dir created where we put different test modules for the documentation so e.g. for this case `tests/unit/documentation/test_changelog.py` - what do you think?

File: tests/unit/test_docs.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import collections
+import os
+import re
+from typing import List
+
+# For entries in the following list, two appearances (but not more) are
+# allowed. This can be useful if a pull request needs to appear in multiple
+# sections.
+ALLOWED_EXCEPTIONS: List[str] = []
```
Comment (valeriupredoi) [inline tests/unit/test_docs.py:10]: quite baffled by the need for this variable :confused:

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
requesting structural changes, nothing code-related from me. Also pinging @bouweandela on this

File: tests/unit/test_docs.py (hunk 1 above)
Comment (schlunma) [reply tests/unit/test_docs.py:1]: Sure, can do that, but note that's the exact location the [corresponding ESMValTool test](https://github.com/ESMValGroup/ESMValTool/blob/main/tests/unit/test_docs.py) lives.

File: tests/unit/test_docs.py (hunk 2 above)
Comment (schlunma) [reply tests/unit/test_docs.py:10]: See discussion here: ESMValGroup/ESMValTool#2554. I thought it might be good to add this feature to this test as well if we need to allow duplicates at some point.

**Comment by zklaus** [comment]
I have to say, I am not so convinced about this approach. Partly because I think it should be enough to have a PR show up in one section. This also aligns with our idea of having exactly one section-related label per PR. Partly because the writing is on the wall that we want to move to a commonly available changelog generator rather than make our little script more elaborate.

**Comment by schlunma** [comment]
My sole intention with this PR was to add a similar test to ESMValCore which was already available in ESMValTool. I was quite surprised that is test isn't available in ESMValCore yet. And it actually found duplicate elements!

I don't want to start a discussion about potential alternatives to our changelog generator here. I just copy-pasted the test from ESMValTool to ESMValCore.

I agree that in a perfect world we don't need to account for duplicates, but I think the argumentation by @axel-lauer in https://github.com/ESMValGroup/ESMValTool/pull/2554 to allow duplicates in certain cases is more than reasonable. This test also doesn't allow duplicates per se, they need to be added manually. Not sure why this is a big deal to be honest.

**Comment by valeriupredoi** [comment]
@zklaus I think this will help with duplicate detection, for now, until we have that modern changelog generator :beer:

**Comment by zklaus** [comment]
Sorry, yes, the test in and of itself is useful and should be added :+1:.
My scepticism is only about the `ALLOWED_DUPLICATES`, which indeed is missing from the corresponding ESMValTool test.

**Comment by schlunma** [comment]
> My scepticism is only about the `ALLOWED_DUPLICATES`, which indeed is missing from the corresponding ESMValTool test.

This is added here: https://github.com/ESMValGroup/ESMValTool/pull/2554. Maybe we should move the discussion to this PR. If we add it there, I suggest we also add it here (and vice versa).

**Comment by valeriupredoi** [comment]
more :baby_chick: :egg: instances :laughing:

**Comment by schlunma** [comment]
I removed the list to allow duplicate entries.

**Review by valeriupredoi — APPROVED** [review]
thought I had approved this, sorry, Manu! OK - we should start thinking how to use Github API for this stuff, in the future, just so we become total nerds :grin:
